### PR TITLE
feat(golangci): add gosec

### PR DESCRIPTION
**Pull request by georgettica**
per ticket  OSD-10161, this change should be running via CI on all osd operators.

this way we are always compliant to the gosec (as we have done one audit a forever ago and cleaned issues

this change should get approval before merging as this might cause initial work to upgrade to this version of boilerplate

### CODE REVIEW

**Comment by 2uasimojo** [comment]
Reasonable.

If you want to preview how this affects CI for a consuming repo before committing to it, you can set Makefile variable `GOLANG_CI_OPTIONAL_CONFIG` ([ref](https://github.com/openshift/boilerplate/blob/b65f892255725d9a566f5aaf50b8c34dd3b38381/boilerplate/openshift/golang-osd-operator/standard.mk#L149)) to point to a local file containing something like:

```yaml
linters:
  disable-all: true
  enable:
  - gosec
```

**Comment by georgettica** [comment]
Oki!
Then let's merge and add this to repo maintainers to adopt

**Comment by 2uasimojo** [comment]
> let's merge and add this to repo maintainers to adopt

I don't understand. Merge what?

**Comment by 2uasimojo** [comment]
Sorry, what I mean is that owners of _consuming_ repos can get an understanding for how this boilerplate PR will affect them when they pull it in -- and perhaps they should do that before we merge this just in case there are show-stopping issues that will take them some time to resolve.

**Comment by 2uasimojo** [comment]
Like [this](https://github.com/openshift/aws-account-operator/pull/685). You can see this will cause `lint` to fail for a-a-o. Proposing (if not merging) a similar PR across all the OSD repositories would be pretty easy, and would alert you in case any of the problems it uncovers end up being really hard to fix. That way you don't get stuck not being able to upgrade boilerplate _past_ this commit in case something more important is merged later.

**Comment by georgettica** [comment]
I agree, and I can do something like that to all the repos so once the checks are fixed, I can promote to boilerplate

**Comment by georgettica** [comment]
/close
as I lost interest :(

**Comment by georgettica** [comment]
/reopen
seems people are talking on it, giving a second try 🤞

**Comment by georgettica** [comment]
related to OSD-11325

**Comment by georgettica** [comment]
I am for merging, and I am not sure I should force merge this.
Maybe someone who is approver can push this forward?

**Comment by 2uasimojo** [comment]
I'll
/approve
but
/hold
to make sure you've properly considered https://github.com/openshift/boilerplate/pull/206#issuecomment-1077797612
TL;DR: if you merge this, consuming repos will be forced to resolve any resulting errors before they can get a bp at *or after* this commit. So like, if you urgently need to pick up an unrelated bp update, you'll be stuck.

File: boilerplate/openshift/golang-osd-operator/golangci.yml [hunk 1]
```diff
@@ -18,6 +18,7 @@ linters:
   enable:
   - deadcode
   - errcheck
+  - gosec
```
Comment (dustman9000) [inline boilerplate/openshift/golang-osd-operator/golangci.yml:21]: We should look at adding rules incrementally to address Eric's concern about all consuming repos failing to pass the lint PR checks once they consume this change. https://golangci-lint.run/usage/linters/#gosec

Comment (dustman9000) [reply boilerplate/openshift/golang-osd-operator/golangci.yml:21]: Maybe just G101-G112 for now? https://github.com/securego/gosec#available-rules

Comment (georgettica) [reply boilerplate/openshift/golang-osd-operator/golangci.yml:21]: Sure, feel free to recreate the PR.

Also I know @Tessg22 is doing the painstakingly hard process of checking per-repo.

Please collab with her if possible on this

**Comment by Tessg22** [comment]
Since [Preview Golangci gosec lint in OSD operators](https://issues.redhat.com/browse/SDE-2056) is complete, we should be ready to unblock this PR

**Comment by NautiluX** [comment]
/hold cancel
All operators cleared issues, this can be merged.
